### DISCUSSION TITLE
About the Licence

### DISCUSSION THREAD

**Discussion by MentalGear** [aspen-cloud/triplit#96] [Q&A]
I've been enjoying exploring triplit quite a lot. It's probably the most elegantly designed lofi fullstack lib I have encountered yet. The only big thing that still holds me back is the 'GNU AFFERO GENERAL PUBLIC LICENSE'.

As far as I understand, since triplit provides both a server and client as a base, this would mean that any work using it/building atop, would need to make its source code available to all users. Please correct me if I'm wrong but this seems to be a possible legal angle that could later come back to bite app devs.

I'm a big fan of open source and giving back, but this kind of license really narrows down the range of adaption and resulting applications that would make sense using triplit for. Was the intention to built triplit only as a solution for open source software?

**Comment by matlin** [comment]
I believe that is a common misconception about AGPL. Our main intention is to ensure that companies that decide to provide Triplit as a service must make any _modifications_ they've made public and available. If you're using Triplit to power your app, my understanding is that you don't need your code to be open-sourced at all. 

Check out this explainer:
https://plausible.io/blog/open-source-licenses#what-is-the-gnu-agplv3-license

**Reply by matlin** [reply]
> I've been enjoying exploring triplit quite a lot. It's probably the most elegantly designed lofi fullstack lib I have encountered yet.

Also so great to hear this!

**Reply by MentalGear** [reply]
You are absolutely right to protect triplit from exploitive behavior like Amazon/AWS, where a monopolistic company adds some propitiatory logic on top and sells it as their own service solution. And I applaud your intention to have it open-source and in a way that others who use it as a service must contribute back.

However, it seems ("_my understanding is_") like you're not absolutely sure how this license could be applied.

If there's ambiguity in the license, and especially to what it applies, there's a good chance a legal interpretation could be made that any apps (not just an expansion of the core service) build on top are also subject to the license.

This makes triplit a really hard sell to use in the techstack when it stands on shaky legal ground, with the constant risk of a competitor suing for the source code and have months or years of app-related work exposed for free. Also consider that some companies outright disallow the use of any kind of software that is AGPL.

I'd advise you to check with your legal council an how to exactly specify to what group of software (app or service) your license applies, and also consider how to present the license in a context (since you said it's a common misconception) that makes it absolutely clear how triplit can be used. (e.g. add specification on top of the license and/or a "Licensing" page on the docs with a simplified bullet list and why you choose AGPL would also be very welcome)

( You might also want to consider the Mongo DB Server-side Public [License](https://www.mongodb.com/legal/licensing/server-side-public-license/faq) )

It's clear by now that I really like triplit and would like to see it prosper, but I'm afraid that for many interested parties as it is, if they check the repo and (first) the license, they might immediately dismiss it without further regard of the excellent tech behind.
